src/algebra/group_with_zero_power.lean

```diff
@@ -0,0 +1,212 @@
+/-
+Copyright (c) 2020 Johan Commelin. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Johan Commelin
+-/
+
+import algebra.group_with_zero
+import algebra.group_power
+import algebra.field
+
+/-!
+# Powers of elements of groups with an adjoined zero element
+
+In this file we define integer power functions for groups with an adjoined zero element.
+This generalises the integer power function on a division ring.
+-/
+
+@[simp] lemma gwz.zero_pow {M : Type*} [monoid_with_zero M] :
+  ∀ n : ℕ, n ≠ 0 → (0 : M)^n = 0
+| 0     h := absurd rfl h
+| (k+1) h := zero_mul _
+
+namespace gwz
```

Comment:
I really don't like `gwz`. Let's write it out.

Reply:
I thought a bit about this, and like I replied to Kevin upstairs, my reasoning was as follows: You probably don't want to open this namespace, because you'll get many conflicting names after that. Hence I chose for a short prefix as namespace.

Reply:
`gwz` is now gone.